src/main/java/net/minecraftforge/event/entity/player/PlayerNegotiationEvent.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.event.entity.player;
+
+import com.mojang.authlib.GameProfile;
+import net.minecraft.network.Connection;
+import net.minecraftforge.eventbus.api.Event;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+/**
+ * This event is fired when a connection has started the Forge Handshake.<br>
+ * Forge will wait for all enqueued work to be completed before proceeding further with the login process.
+ */
```

Comment:
Some more documentation here would be great.
What side does this event fire on? Which bus?

Document the use-case here too, please.

Reply:
I've added the requested documentation